### ISSUE TITLE
actives = new WeakMap();

### ISSUE THREAD

**Issue by acandael** [btholt/complete-intro-to-containers#8]
I'm stuck at the lesson 'Add Dependencies to a Node.js App. I have installed @hapi/hapi and hapi-pino, but when I run: node index.js I get this error message:

/Users/acandael/Tutorials/intro-to-containers/more-complicated-nodejs-app/node_modules/@hapi/hapi/lib/core.js:51
    actives = new WeakMap();   

SyntaxError: Unexpected token =

my version of node is 10.16.3
my version of @hapi/hapi is 19.0.5

anyone else having this error message?

**Comment by KayacChang** [comment]
Same as I

```
/home/node/code/node_modules/@hapi/hapi/lib/core.js:51
    actives = new WeakMap();                                                   // Active requests being processed
            ^

SyntaxError: Unexpected token =
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/node/code/node_modules/@hapi/hapi/lib/server.js:9:14)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
```

**Comment by KayacChang** [comment]
Change `RUN apk add --update nodejs` to `RUN apk add --update nodejs-current`

will fix this issue.

```
# build stage
FROM node:12-stretch AS build

WORKDIR /build

COPY --chown=node:node package-lock.json package.json ./

RUN npm ci --production

COPY . .

# runtime stage
FROM alpine:3.10

RUN apk add nodejs-current

RUN addgroup -S node && adduser -S node -G node

USER node

RUN mkdir /home/node/code

WORKDIR /home/node/code

COPY --from=build --chown=node:node /build .

CMD ["node", "index.js"]

```

**Comment by acandael** [comment]
Thanks KayaChang. I ran into the same issue at the lesson 'Alpine Node.js Container.
In the Dockerfile I put the line:
--update nodejs-current npm
and then I could run the container.

**Comment by btholt** [comment]
cc @1Marc 

That's bizarre. I just tried this again and it still works for me. Somehow you all have to be getting Node.js version 8 inside of your container which shouldn't be happening.

Marc, can we add an annotation that says something like "If you're seeing errors in the JavaScript, try using `nodejs-current` instead of `nodejs`? I can add a note in the class notes too.

Thank you @acandael @KayacChang!

**Comment by luisfta17** [comment]
Hi @btholt, 
Ran into the exact same issue as @acandael and adding "--update nodejs-current npm" also fixed the issue for me!

**Comment by malcolm-kee** [comment]
For me I have the same issue **not** in the container, but when I tried to run it in my machine.
It's fixed by upgrading nodejs version.

**Comment by nbckr** [comment]
Just to give you some more data, this is what I found out. Following the course (which is fantastic btw, thanks!), using `alpine:3.10` as base image and running  `apk add --update nodejs`:

```bash
docker run --rm --init -p 3000:3000 fem:6 node -v
v10.19.0
```

It seems to be an issue with hapi and the node version indeed. When I use node 13.0.1 locally, all works fine. Using an older version however allows me to recreate the issue:

```bash
nvm use 10.10.0
Now using node v10.10.0 (npm v6.4.1)

» node index.js
/Users/nibo/Projects/frontendmasters/complete-intro-to-containers/multi-stage/node_modules/@hapi/hapi/lib/core.js:51
    actives = new WeakMap();                                                   // Active requests being processed
            ^

SyntaxError: Unexpected token =
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:657:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:20:18)
```

**Comment by btholt** [comment]
Fixed by #13
